EIPS/eip-5222.md

```diff
@@ -0,0 +1,237 @@
+
+## Backwards Compatibility
+
+This standard is compatible with the current EIP-721 standards: a contract can inherit from both EIP-721 and EIP-5222 at the same time.
+
+## Test Cases
+
+Test cases are available [here](../assets/eip-5222/contracts/test/Contract.t.sol).
+
+## Reference Implementation
+
+A reference implementation maintains the following data structures:
+
+```solidity
+  struct License {
+    bool active; // whether the license is active
+    uint256 tokenId; // the identifier of the NFT the license is tethered to
+    uint256 parentLicenseId; // the identifier of the parent license
+    address licenseHolder; // the license holder
+    string uri; // the license URI
+    address revoker; // the license revoker
+  }
+  mapping(uint256 => License) private _licenses; // maps from a license identifier to a license object
+  mapping(uint256 => uint256) private _licenseIds; // maps from an NFT to its root license identifier
+```
+
+Each NFT has a license tree and starting from each license, one can trace back to the root license via `parentLicenseId` along the path. 
+
+In the reference implementation, once a license is revoked, all sublicenses under it are revoked. This is realized in a *lazy* manner for lower gas cost, i.e., assign `active=false` only for licenses that are explicitly revoked in a `revokeLicense` function call. Therefore, `isLicenseActive` returns `true` only if all its ancestral licenses haven't been revoked.
+
+For non-root licenses, the creation, transfer and revocation are straightforward:
+
+1. Only the holder of an active license can create sublicenses.
+2. Only the holder of an active license can transfer it to a different license holder. 
+3. Only the revoker of an active license can revoke it.
+
+The root license must be compatible with `EIP-721`: 
+
+1. When an NFT is minted, a license is granted to the NFT owner.
+2. When an NFT is transferred, the license holder is changed to the new owner of the NFT.
+3. When a root license is revoked, the NFT is returned to the NFT creator, and the NFT creator may later transfer it to a new owner with a new license.
+
+The complete implementation can be found [here](../assets/eip-5222/contracts/src/RightsManagement.sol). 
+
+In addition, the [IC3 NFT License](../assets/eip-5222/ic3license/ic3license.pdf) is specifically designed to work with this interface and provides a reference to the language of NFT licenses.
```

Comment:
(Disregard my deleted comment. For some reason I thought this was an external link.)